### ISSUE TITLE
Get an item from an array or object using "dot" notation

### ISSUE THREAD

**Issue by JCWolf** [ecrmnn/collect.js#177]
Some methods, like `pluck()` for example should accept dot notation keys. 

This is the [Laravel equivalent](https://github.com/tightenco/collect/blob/master/src/Collect/Support/Arr.php#L384) for `pluck` and this is for [dot notation get](https://github.com/tightenco/collect/blob/master/src/Collect/Support/helpers.php#L47).

For example, having this collection:
```ts
let users = [
    {
        "name": "John",
        "roles": [
                {"name": "Editor"},
                {"name": "Manager"}
        ]
    }
]
```

After calling pluck: `let allRoles = collect(users).pluck('roles.*.name')`; should return;

```ts
[
   ["Editor","Manager"]
]

```

_PS. And of course, to get just the roles we have to do `.flatten(1)` after pluck but this is not the topic here. :)_

**Comment by ecrmnn** [comment]
I'm not getting the same results. Could you provide your PHP example?

```php
>>> $users = [
    'name' => 'John',
    'roles'=> [
        [
            'name' => 'Editor',
        ],
        [
            'name' => 'Admin',
        ],
    ],
];

>>> collect($users)->pluck('roles.*.name');
=> Illuminate\Support\Collection {#753
     all: [
       null,
       null,
     ],
   }
>>>

**Comment by JCWolf** [comment]
`$users` should be a collection of users and not a single user, like so:
```php
$users = [
	[
	    'name' => 'John',
	    'roles'=> [
	        ['name' => 'Editor'],
	        ['name' => 'Admin'],
	    ],
	]
];
```

**Comment by ecrmnn** [comment]
Yeah, I get the same as you now. I'll implement it asap, but in the meantime a PR is welcomed.

Do you know what other methods that allow dot notation? 

```php
>>> collect($users)->get('roles.1.name')
=> null
```

**Comment by JCWolf** [comment]
Yes, I know some methods like:
- [get('some.deep.key')](https://github.com/tightenco/collect/blob/master/src/Collect/Support/Arr.php#L278)
- [forget('some.deep.key')](https://github.com/tightenco/collect/blob/master/src/Collect/Support/Arr.php#L233)
- [has('some.deep.key')](https://github.com/tightenco/collect/blob/master/src/Collect/Support/Arr.php#L314)

I never tested these above methods. I was just looking for the keyword `explode` inside [Arr class](https://github.com/tightenco/collect/blob/master/src/Collect/Support/Arr.php). They seem to have the same behaviour.

**Comment by JCWolf** [comment]
The `get`, `forget` and `has` methods operate for objects while `pluck` operates for arrays.

Try this:
```php
$user = [
        "name"=> "John",
        "status" => [
            "name" => "Active"    
        ]
];

collect($users)->get('status.name');
```

To make it work with my original example you have to write it like so: 
```php
collect($users)->get('0.roles.1.name')
```

**Comment by ecrmnn** [comment]
I'm getting this with `php artisan tinker` on a fresh `5.6` installation

```php
>>> $user = [
...         "name"=> "John",
...         "status" => [
...             "name" => "Active"
...         ]
... ];

=> [
     "name" => "John",
     "status" => [
       "name" => "Active",
     ],
   ]

>>> collect($user)->get('status.name');
=> null

>>> collect($user)->has('status.name');
=> false
>>>
```

**Comment by JCWolf** [comment]
Check and [run this](https://3v4l.org/0sJAH), please. I copied the entire `Arr` class and it works as expected(the example is at the bottom).

**Comment by JCWolf** [comment]
I checked the [Collection](https://github.com/tightenco/collect/blob/master/src/Collect/Support/Collection.php#L716)  code and you are right. They are not using the `Arr::get` method inside the `Collection::get`. They only one that works is `pluck` but would be nice to have the others too.

**Comment by ecrmnn** [comment]
Update; I’m working on getting this implemented. My time is limited these days so I can’t say exactly when I’m done
